hooks/checkout

```diff
@@ -40,6 +40,36 @@ check_set() {
   fi
 }
 
+
+# Works like GNU timeout, but implementing it here lets it run on macOS agents.
+# Mostly copied from here: https://stackoverflow.com/a/24413646
+timeout() {
+  local -r duration=$1
+  shift
+  local -r cmd=( "$@" )
+
+  # Run in a subshell to avoid job control messages
+  (
+    "${cmd[@]}" &
+    child=$!
+    # Avoid default notification in non-interactive shell for SIGTERM
+    trap -- "" SIGTERM
+    (
+      sleep "$duration"
+      kill "$child" 2> /dev/null
+    ) &
+    wait "$child"
+  )
+
+  # Return 124 if a timeout occurred, or the exit code of the command.
+  local -r rc=$?
+  if [[ "$rc" -eq 143 ]]; then
+    return 124
+  else
+    return "$rc"
+  fi
+}
```

Comment:
You could argue that this should be named something other than "timeout" to differentiate it from GNU's `timeout`. I think it's fine as long as they work similarly enough. Hence, rewriting the timeout exit code to be more like GNU's.